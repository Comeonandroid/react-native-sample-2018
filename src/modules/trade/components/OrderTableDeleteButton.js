import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { prop } from 'styled-tools'
import PropTypes from 'prop-types'

import { View } from '../../../elements'
import CustomIcon from '../../../utils/icon'

class OrderTableDeleteButton extends PureComponent {

  constructor(props) {
    super(props)
    this.handlePress = this.handlePress.bind(this)
  }

  handlePress() {
    this.props.onPress(this.props.index)
  }

  render() {
    return (
      <ButtonBackground>
        <Button onPress={this.handlePress}>
          <Icon name='cancel' size={11} />
        </Button>
      </ButtonBackground>
    )
  }
}

OrderTableDeleteButton.propTypes = {
  onPress: PropTypes.func,
  index: PropTypes.number
}

const ButtonBackground = styled(View) `
  height: 32px;
  width: 32px;
  padding: 0 5px;
  display: flex;
  justify-content: center;
  align-self: center;
`

const Button = styled.TouchableOpacity`
  background: ${prop('theme.colors.danger')}
  padding: 5px;
  borderRadius: 20px;
`

const Icon = styled(CustomIcon) `
  color: ${prop('theme.colors.background')}
`


export default OrderTableDeleteButton
