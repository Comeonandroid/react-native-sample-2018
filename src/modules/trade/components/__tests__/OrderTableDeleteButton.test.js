import 'react-native'
import React from 'react'

import { itBehavesLike } from '../../../../specs/shared_examples/index'

import OrderTableDeleteButton from '../OrderTableDeleteButton'

const setup = propOverrides => {
  const props = Object.assign({
    data: [
      ['table line 1'],
      ['table line 2'],
      ['table line 3']
    ]
  }, propOverrides)

  const component = <OrderTableDeleteButton {...props} />

  return {
    props,
    component
  }
}

describe('<OrderTableDeleteButton />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })
})
