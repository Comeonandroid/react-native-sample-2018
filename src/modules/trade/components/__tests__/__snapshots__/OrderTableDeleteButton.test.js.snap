// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderTableDeleteButton /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignSelf": "center",
          "display": "flex",
          "height": 32,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 0,
          "width": 32,
        },
        undefined,
      ],
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f25c74",
        "borderRadius": 20,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 11,
          },
          Array [
            Object {
              "color": "#fcfcfc",
            },
            Array [
              Object {
                "color": "#000000",
              },
              undefined,
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`<OrderTableDeleteButton /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignSelf": "center",
          "display": "flex",
          "height": 32,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 0,
          "width": 32,
        },
        undefined,
      ],
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f25c74",
        "borderRadius": 20,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 11,
          },
          Array [
            Object {
              "color": "#4b4c4d",
            },
            Array [
              Object {
                "color": "#fefaec",
              },
              undefined,
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
